test/testdata/packager/unimported_namespace/aaa/a_class.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+# typed: strict
+
+class AAA::AClass
+  BBB
+# ^^^ error: Unable to resolve constant `BBB`
```

Comment:
Prior to this change, this would have resolved. That's the whole point of this change.